libs/platform/experience/entity-text/entity-text.component.spec.ts

```diff
@@ -1,77 +1,138 @@
+      );
+    });
+
+    it('should call the entity service to resolve the data', () => {
+      expect(entityService.getField).toHaveBeenCalledWith({
+        element: element,
+        type: 'data',
+        field: 'name',
+      });
+    });
   });
 
-  it('passes the a11y audit', async () => {
-    element = await fixture(
-      html`<oryx-product-title
-        sku="1"
-        .options=${{ tag: 'h1' }}
-      ></oryx-product-title>`
-    );
+  describe('when text is resolved for the field', () => {
+    beforeEach(async () => {
+      entityService.getField.mockReturnValue('foo bar');
+      element = await fixture(
+        html`<oryx-entity-text
+          .options=${{ entity: 'data', field: 'name' }}
+        ></oryx-entity-text>`
+      );
+    });
+
+    it('should render the text', () => {
+      expect(element.shadowRoot?.textContent).toContain('foo bar');
+    });
 
-    await expect(element).shadowDom.to.be.accessible();
+    describe('and a prefix option is configured', () => {
+      beforeEach(async () => {
+        element = await fixture(
+          html`<oryx-entity-text
+            .options=${{ entity: 'data', field: 'name', prefix: 'prefix-' }}
+          ></oryx-entity-text>`
+        );
+      });
+
+      it('should render the prefix', () => {
+        expect(element.shadowRoot?.textContent?.trim()).toEqual(
+          'prefix-foo bar'
+        );
+      });
+    });
   });
 
-  describe('when the entity type is product and field is name', () => {
+  describe('when a tag option is configured', () => {
     beforeEach(async () => {
+      entityService.getField.mockReturnValue('foo bar');
       element = await fixture(
         html`<oryx-entity-text
-          .options=${{ entity: 'product', field: 'name' }}
+          .options=${{ entity: 'data', field: 'name', tag: HeadingTag.H3 }}
         ></oryx-entity-text>`
       );
     });
 
-    it('should render a product name', () => {
-      expect(element.shadowRoot?.innerHTML).toContain('Sample product');
+    it('should render the heading component', () => {
+      expect(element).toContainElement('oryx-heading');
+    });
+
+    it('should render text in the heading component', () => {
+      expect(element.shadowRoot?.textContent?.trim()).toEqual('foo bar');
+    });
+
+    describe('and a prefix option is configured', () => {
+      beforeEach(async () => {
+        element = await fixture(
+          html`<oryx-entity-text
+            .options=${{
+              entity: 'data',
+              field: 'name',
+              tag: HeadingTag.H3,
+              prefix: 'prefix-',
+            }}
+          ></oryx-entity-text>`
+        );
+      });
+
+      it('should render the prefix inside the heading', () => {
+        const heading = element.shadowRoot?.querySelector('oryx-heading');
+        expect(heading?.textContent?.trim()).toEqual('prefix-foo bar');
+      });
     });
   });
+
+  // describe('when a link option is configured', () => {
```

Comment:
Left over?

Reply:
yeah, we'll add link soon, hence we kept a few left overs for that